pkg/codegen/python/gen.go

```diff
@@ -624,9 +611,26 @@ func (mod *modContext) genAwaitableType(w io.Writer, obj *schema.ObjectType) str
 			fmt.Fprintf(w, "            pulumi.log.warn(\"%s is deprecated: %s\")\n\n", pname, escaped)
 		}
 
-		// Now perform the assignment, and follow it with a """ doc comment if there was one found.
-		fmt.Fprintf(w, "        __self__.%[1]s = %[1]s\n", pname)
-		printComment(w, prop.Comment, "        ")
+		// Now perform the assignment.
+		fmt.Fprintf(w, "        pulumi.set(__self__, \"%[1]s\", %[1]s)\n", pname)
+	}
+	fmt.Fprintf(w, "\n")
+
+	// Write out Python property getters for each property.
+	for _, prop := range obj.Properties {
+		pname := PyName(prop.Name)
+		ty := mod.typeString(prop.Type, false /*input*/, false /*wrapInput*/, !prop.IsRequired, false /*acceptMapping*/)
+		fmt.Fprintf(w, "    @property\n")
+		if pname == prop.Name {
+			fmt.Fprintf(w, "    @pulumi.getter\n")
+		} else {
+			fmt.Fprintf(w, "    @pulumi.getter(name=%q)\n", prop.Name)
+		}
```

Comment:
Do we still need the decorator if we're emitting the function bodies?

Reply:
Yes. It's needed to provide the mapping between the Python name and Pulumi name.